aea/cli/generate.py

```diff
@@ -119,17 +156,51 @@ def _generate_item(ctx: Context, item_type: str, specification_path: str) -> Non
     except FileExistsError:
         raise click.ClickException(  # pragma: no cover
             "A {} with this name already exists. Please choose a different name and try again.".format(
-                item_type
+                PROTOCOL
             )
         )
-    except ProtocolSpecificationParseError as e:
-        raise click.ClickException(  # pragma: no cover
-            "The following error happened while parsing the protocol specification: "
+    except Exception as e:
+        raise click.ClickException(
+            "Protocol is NOT generated. The following error happened while generating the protocol:\n"
             + str(e)
         )
+    fingerprint_item(ctx, PROTOCOL, protocol_spec.public_id)
+
+
+@clean_after
+def _generate_protobuf_mode(
+    ctx: Context,
+    protocol_generator: ProtocolGenerator,
+    protocol_spec: ProtocolSpecification,
+    existing_id_list: Set[PublicId],
+    language: str,
+) -> None:
+    """Generate a protocol in 'protobuf' mode, and add it to the configuration file and agent."""
+    try:
+        warning_message = protocol_generator.generate(
+            protobuf_only=True, language=language
+        )
+        if warning_message is not None:
+            click.echo(warning_message)
+
+        # Add the item to the configurations
+        logger.debug(
+            "Registering the {} into {}".format(PROTOCOL, DEFAULT_AEA_CONFIG_FILE)
+        )
+        existing_id_list.add(
+            PublicId(protocol_spec.author, protocol_spec.name, protocol_spec.version)
+        )
+        ctx.agent_loader.dump(
+            ctx.agent_config, open(os.path.join(ctx.cwd, DEFAULT_AEA_CONFIG_FILE), "w")
+        )
```

Comment:
No point in adding it to the agent. Only Python packages should be added.

Reply:
Removed